webapp-django/crashstats/crashstats/models.py

```diff
@@ -48,6 +51,8 @@ def config_from_configman():
         definition_source=[
             ElasticsearchConfig.required_config,
             PostgreSQLCrashStorage.required_config,
+            ReprocessingOneRabbitMQCrashStore.required_config,
```

Comment:
_ah ha!_  these are the problem, not the change_default that we discussed earlier.  

1) In each one of these lines, you're referencing the class var ```required_config``` of the derived class.  This ignores the ```required_config``` of any base classes.  The quick fix is to replace the ```.required_config``` with ```.get_required_config()```.  That latter function walks the inheritance hierarchy to get a union of all the required config from every level.

2) however, you've got a second problem.  By having both PG and RMQ at the same level you're going to have name collisions in the options (both have 'transaction_executor_class', for example)  Each of these ought to be in namespaces.  Do something like this:

``` python
definition_source = Namespace()
definition_source.namespace('search')
definition_source.search.add_option(
     'search_class',
     default='socorro.external.es.base.ElasticsearchConfig',
     from_string_converter=class_converter
)
definition_source.namespace('database')
definition_source.database.add_option(
     'database_storage_class',
     default='socorro.external.postgresql.crashstorage.PostgreSQLCrashStorage',
     from_string_converter=class_converter
)
definition_source.namespace('queuing')
definition_source.queuing.add_option(
     'queuing_class',
     default='socorro.external.rabbitmq.crashstorage.ReprocessingOneRabbitMQCrashStore',
     from_string_converter=class_converter
)
```
Anywhere that you reference these values, you then should refer to them with their namespaces:

``` python
# to create the ES object
search_object = config.search.search_class(config.search)
...
```